boards/components/src/i2c.rs

```diff
@@ -52,8 +54,14 @@ pub struct I2CComponent {
 }
 
 impl I2CMuxComponent {
-    pub fn new(i2c: &'static dyn i2c::I2CMaster) -> Self {
-        I2CMuxComponent { i2c: i2c }
+    pub fn new(
+        i2c: &'static dyn i2c::I2CMaster,
+        deferred_caller: &'static DynamicDeferredCall,
```

Comment:
Should this not also take in a `smbus: Option<&'a dyn i2c::SMBusMaster>` and pipe it to the MuxI2C::new call?

Reply:
It would be fine to leave this as-is if you think the plan will be to have a different component for SMBus use.

Reply:
Yes, I have fixed this. The board can specify to use the same as I2C or something else